app/assets/javascripts/course/assessment/submission/answer/programming.js

```diff
@@ -216,6 +216,33 @@
   }
 
   /**
+   * Creates a annotation form for the user to edit his annotation post.
+   *
+   * @param {jQuery} $element The element to search for the form.
+   * @param {Number} courseId The course ID that the annotation is associated ith.
+   * @param {Number} assessmentId The assessment ID that the annotation is associated with.
+   * @param {Number} submissionId The submission ID that the annotation is associated with.
+   * @param {Number} answerId The answer ID that the annotation is associated with.
+   * @param {Number} programmingFileId The programming answer file ID that the annotation is
+   *   associated with.
+   * @param {Number} lineNumber The line number that the user is annotating.
+   * @param {Number} postId The ID of the post being edited.
+   * @param {String} postContent The existing contents of the post's text field.
+   * @param {HTMLElement} postCommenter The commenter's name with a link to his profile.
+   * @return {jQuery} The annotation editing form which was found or created.
+   */
+  function findOrCreateAnnotationPostForm($element, courseId, assessmentId, submissionId, answerId,
```

Comment:
I suggest you take a look at the js code of comments center, to see if this can be reused.